### PR TITLE
[ML] renames */inference* apis to */trained_models*

### PR DESCRIPTION
**Pull request by benwtrent**
This commit renames all `inference` CRUD APIs to `trained_models`.

This aligns with internal terminology, documentation, and use-cases.

### CODE REVIEW

File: docs/reference/aggregations/pipeline/inference-bucket-aggregation.asciidoc [hunk 1]
```diff
@@ -53,8 +53,8 @@ See <<buckets-path-syntax>> for more details | Required       | -
 
 ==== Configuration options for {infer} models
 
-The `inference_config` setting is optional and usually isn't required as the 
-pre-trained models come equipped with sensible defaults. In the context of 
+The `inference_config` setting is optional and usually isn't required as the
```
Comment (przemekwitek) [inline docs/reference/aggregations/pipeline/inference-bucket-aggregation.asciidoc:56]: Is the ultimate plan to also rename the `inference_config` field?

**Review by przemekwitek — APPROVED** [review]
LGTM

**Review by szabosteve — APPROVED** [review]
Thanks for the changes and for even taking care of the redirects! LGTM.

**Comment by benwtrent** [comment]
run elasticsearch-ci/2
run elasticsearch-ci/packaging-sample-windows

File: docs/reference/aggregations/pipeline/inference-bucket-aggregation.asciidoc (hunk 1 above)
Comment (benwtrent) [reply docs/reference/aggregations/pipeline/inference-bucket-aggregation.asciidoc:56]: no. `inference_config` is the default configuration when using the model during inference.

File: docs/reference/redirects.asciidoc [hunk 2]
```diff
@@ -540,6 +540,26 @@ See <<put-transform>>, <<preview-transform>>, <<update-transform>>,
 This page was deleted.
 See <<put-dfanalytics>>.
 
+[role="exclude",id="put-inference"]
```
Comment (davidkyle) [inline docs/reference/redirects.asciidoc:543]: ▶️ 👍

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**